osi_logicallane.proto

```diff
@@ -0,0 +1,805 @@
+    // - If there are several lane boundaries, they must be listed in increasing
+    //   T order (i.e. from right to left in reference line direction).
+    //   Rationale: this makes it easier to determine e.g. rules on lane
+    //   changes, which depend on the T order of the lanes.
+    // - Whenever physical lane boundaries begin or end, or switch their T
+    //   position (if there are multiple physical lane boundaries), a new
+    //   LogicalLaneBoundary must be created.
+    // - The referenced LaneBoundary objects may be longer than the
+    //   LogicalLaneBoundary which references them, but must never be shorter.
+    //
+    // Example:
+    // <pre>
+    //       Lane 1
+    //  --------a------------------ - - - -c- - - -           ----&gt;
+    //  \- - - -b- - - -
+    //       Lane -1
+    // </pre>
+    //
+    // This shows the boundary between lane 1 and lane -1, with the reference
+    // line going from left to right. First there is a solid-broken line (a and
+    // b), then there is only a solid line (a), then there is a broken line
+    // (c). There would be three LogicalLaneBoundary objects between Lane1 and
+    // Lane2: the first would reference first b and then a, the second would
+    // reference only a, and the third would reference c.
+    //
+    // \rules
+    // refers_to: LaneBoundary
+    // \endrules
+    //
+    repeated Identifier physical_boundary_id = 4;
+
+    // The passing rules, insomuch as they can be determined just from road
+    // markings.
+    //
+    optional PassingRule passing_rule = 5;
+
+    // Optional external reference to the lane boundary source.
+    //
+    // \note For OpenDRIVE, there is no direct possibility to reference the
+    //       RoadMark, as there is no unique identifier in this sub-object.
+    //
+    // \note For non-ASAM Standards, it is implementation-specific how
+    //       source_reference is resolved.
+    //
+    // \note The value has to be repeated because one object may be derived
+    //       from more than one origin source, for example, from a scenario file
+    //       and from sensors.
+    //
+    repeated ExternalReference source_reference = 6;
+
+    //
+    // \brief A point on the boundary
+    //
+    message LogicalBoundaryPoint
+    {
+
+        // The position of the \c LaneBoundaryPoint.
+        //
+        optional Vector3d position = 1;
+
+        // S position of the LaneBoundaryPoint, measured along the parent's
+        // reference_line_id.
+        //
+        optional double s_position = 2;
+
+        // T position of the LaneBoundaryPoint, measured along the parent's
+        // reference_line_id.
+        //
+        optional double t_position = 3;
+    }
+
+    //
+    // Passing rule of the LogicalLaneBoundary.
+    //
+    // This describes how vehicles may move across the LogicalLaneBoundary. The
+    // PassingRule is determined solely based on the road, not on any signs
+    // (i.e. it may be overridden by signs).
```

Comment:
```suggestion
    // This describes how vehicles are legally allowed to move across the 
    // LogicalLaneBoundary. The PassingRule is determined solely based 
    // on the semantics of physical road markings, not on any signs
    // (i.e. it may be overridden by signs).
```